File: FunnyAlgorithms-master/PalindromeChecker/PalindromeChecker.cpp
#include<iostream>
using namespace std;
bool isPalindrome(string s)
{
	for(int i=0;i<s.size()/2;i++)
	{
		if(s[i]!=s[s.size()-i-1])
		{
			return false;
		}
	}
	return true;
}

int main()
{
	string s;
	cin>>s;
	if(isPalindrome(s))
		cout<<"String \""<<s<<"\" is a palindrome";
	else
		cout<<"String \""<<s<<"\" is not a palindrome";
}

File: FunnyAlgorithms-master/PalindromeChecker/PalindromeChecker.go
package main

import (
	"fmt"
	"strings"
)

func isPalindrome(word string) bool {
	word = strings.ToLower(word)

	for i := 0; i < len(word)/2; i++ {
		if word[i] != word[len(word)-i-1] {
			return false
		}
	}
	return true
}

func main() {
	var word string
	fmt.Print("Input a word: ")
	fmt.Scan(&word)

	if !isPalindrome(word) {
		fmt.Println("Sorry, that's not a palindrome")
	} else {
		fmt.Println("It's a palindrome")
	}
}
